src/main/webapp/js/components/csv/csv-config-page.jsx

```diff
@@ -0,0 +1,285 @@
+                                            { _.map(this.getRelatedFields(fieldConfig.relatedField, fieldConfig.relatedEntity), relatedElem => renderFormField({ ...relatedElem, name: `${relatedName}.${relatedElem.name}` })) }
+                                          </div>
+                                        ))}
+                                        <div>
+                                          <button
+                                            type="button"
+                                            className="btn btn-success my-2"
+                                            onClick={() => relatedFields.push({
+                                              relatedField: '',
+                                              relatedFieldCsvName: '',
+                                            })}
+                                          >
+                                                Add related field
+                                          </button>
+                                        </div>
+                                      </div>
+                                    )}
+                                  </FieldArray>
+                                </Field>
+                              </div>
+                            );
+                          }
+                          return null;
+                        })()}
+                        <button
+                          type="button"
+                          className="btn btn-outline-danger"
+                          onClick={() => fields.remove(index)}
+                        >
+                          Delete
+                        </button>
+                      </div>
+                    ))}
+                    <button
+                      type="button"
+                      className="btn btn-outline-success mr-2"
+                      disabled={!entityType}
+                      onClick={() => {
+                        fields.push({
+                          fieldConfigId: '',
+                          fieldName: '',
+                          format: '',
+                          defaultValue: '',
+                        });
+                      }}
+                    >
+                      Add field config
+                    </button>
+                    <button
+                      type="submit"
+                      className="btn btn-outline-primary m-2"
+                      disabled={invalid || !entityType}
+                      /* eslint-disable-next-line max-len */
+                      onClick={() => { this.onSubmit(values.csvFields); }}
+                    >
+                      Save
+                    </button>
+                    <button
+                      type="button"
+                      className="btn btn-outline-danger m-2"
+                      disabled={!entityType}
+                      onClick={() => { this.deleteConfig(); }}
```

Comment:
if this button removing the entire config why it's in the field array? I think this should be move to the parent